core/tx_pool.go

```diff
@@ -1210,7 +1242,11 @@ func (pool *TxPool) runReorg(done chan struct{}, reset *txpoolResetRequest, dirt
 	if len(events) > 0 {
 		var txs []*types.Transaction
 		for _, set := range events {
-			txs = append(txs, set.Flatten()...)
+			for _, tx := range set.Flatten() {
+				if !pool.IsPrivateTxHash(tx.Hash()) {
+					txs = append(txs, tx)
+				}
+			}
```

Comment:
Is this really needed? As long as the private ones are synced with a peer, isn't that enough? This feed might be used by other subsystems, do you really need to prevent them going out there too? 
Just checking

Reply:
I think the main transaction broadcast loop uses this feed to broadcast to peers, so we need to avoid that going out in some way:

```go
func (h *handler) Start(maxPeers int) {
	h.maxPeers = maxPeers

	// broadcast transactions
	h.wg.Add(1)
	h.txsCh = make(chan core.NewTxsEvent, txChanSize)
	h.txsSub = h.txpool.SubscribeNewTxsEvent(h.txsCh)
	go h.txBroadcastLoop()
...
```

Other subsystems may or may not be ok, but I think it's not completely clear cut (`newPendingTxs` doesn't really matter to most of the users of nodes running this patch, for instance).